lbclient.go

```diff
@@ -98,6 +98,18 @@ func (cc *LBClient) init() {
 	// hammering from a cluster of identical LBClients.
 	cc.nextIdx = uint32(time.Now().UnixNano())
 }
+// clients update: add & remove
+func (cc *LBClient) AddClients(c *HostClient) int {
+    cc.cs = append(cc.cs, &lbClient{
+        c:           c,
+        healthCheck: cc.HealthCheck,
+    })
```

Comment:
This is not thread safe. `cc.cs` can be accessed from multiple goroutines. This will require a `sync.Mutex`.

Reply:
sync.Mutex will kill the performance, we should think of alternative solutions first

Reply:
No it won't. A slice get guarded by a Lock and Unlock only takes around 22 nanoseconds. Any network operation is going to be multiple magnitudes of orders slower. Unless you're doing hundreds of thousands of Lock/Unlock operations per second optimizing a Mutex makes no sense at all.